frontend/www/js/omegaup/components/user/ManageApiTokens.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div>
+    <form class="form add-identity-form" @submit.prevent="$emit('create-api-token', tokenName)">
+      <div class="form-group">
+        <p>
+          {{ T.apiTokenDescription }}
+        </p>
+      </div>
+      <div class="form-group">
+        <label class="w-100">
+          {{ T.wordsTokenName }}
+          <font-awesome-icon :title="T.profileAddIdentitiesTooltip" icon="info-circle"></font-awesome-icon>
+          <input v-model="tokenName" autocomplete="off" class="form-control username-input" size="20" type="text" />
+        </label>
+      </div>
+      <div class="form-group text-right">
+        <button class="btn btn-primary" type="submit">
+          {{ T.wordsAddApiToken }}
+        </button>
+      </div>
+    </form>
+    <div v-if="apiTokens.length == 0">
+      <div class="empty-category">
+        {{ T.profileIdentitiesEmpty }}
+      </div>
+    </div>
+    <table v-else class="table table-striped table-over">
+      <thead>
+        <tr>
+          <th>{{ T.wordsTokenName }}</th>
+          <th>{{ T.timestamp }}</th>
+          <th>{{ T.lastTimeUsed }}</th>
+          <th>{{ T.resetTime }}</th>
+          <th>{{ T.wordsRemaining }}</th>
+          <th>{{ T.wordsLimit }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="apiToken in apiTokens" :key="apiToken.name">
+          <td>{{ apiToken.name }}</td>
+          <td>{{ formatTime(apiToken.timestamp) }}</td>
+          <td>{{ formatTime(apiToken.last_used) }}</td>
+          <td v-if="apiToken.rate_limit && apiToken.rate_limit.reset">{{ formatTime(apiToken.rate_limit.reset) }}</td>
+          <td v-if="apiToken.rate_limit && apiToken.rate_limit.remaining">{{ apiToken.rate_limit.remaining }}</td>
+          <td v-if="apiToken.rate_limit && apiToken.rate_limit.limit">{{ apiToken.rate_limit.limit }}</td>
+          <td>
+            <button class="btn btn-secondary" @click="$emit('revoke-api-token', apiToken.name)">
+              {{ T.revokeApiToken }}
+            </button>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import T from '../../lang';
+import * as time from '../../time';
+
+import {
+  FontAwesomeIcon,
+  FontAwesomeLayers,
+  FontAwesomeLayersText,
+} from '@fortawesome/vue-fontawesome';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+import { library } from '@fortawesome/fontawesome-svg-core';
+library.add(fas);
+
+@Component({
+  components: {
+    'font-awesome-icon': FontAwesomeIcon,
+    'font-awesome-layers': FontAwesomeLayers,
+    'font-awesome-layers-text': FontAwesomeLayersText,
+  },
+})
+export default class ManageApiTokens extends Vue {
+  @Prop() apiTokens!: types.ApiToken[];
```

Comment:
nit:
```suggestion
  @Prop({ default: () => [] }) apiTokens!: types.ApiToken[];
```
Acá le podemos definir un valor de array vacío por defecto para evitar que nos marque errores en lugares cuando no recibamos nada, por ejemplo en las pruebas de jest